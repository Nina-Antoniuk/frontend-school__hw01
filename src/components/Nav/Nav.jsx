import React from 'react';
import { NavLink } from 'react-router-dom';
import s from './Nav.module.scss';

const Nav = function Nav() {
  return (
    <nav className={s.Nav}>
      <ul className={s.NavList}>
        <li>
          <NavLink
            exact
            to="/"
            className={s.NavLink}
            activeClassName={s.activeLink}
          >
            News
          </NavLink>
        </li>
        <li>
          <NavLink
            to={`/profile/${'dave.xp'}`}
            className={s.NavLink}
            activeClassName={s.activeLink}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
export default Nav;
