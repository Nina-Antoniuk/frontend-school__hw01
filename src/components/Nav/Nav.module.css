@media screen and (min-width: 768px) {
  .Nav {
    position: fixed;
    min-width: 20%;
    height: 100vh;
    padding: 3vh 2vw;
  }
}

.NavList {
  padding: 0;
  list-style-type: none;
}

@media screen and (max-width: 767.98px) {
  .NavList {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-pack: distribute;
    justify-content: space-around;
  }
}

.NavLink {
  display: block;
  padding: 10px 0;
  font-size: 20px;
  text-decoration: none;
  color: var(--dark-color);
  -webkit-transition: color var(--delay) ease;
  transition: color var(--delay) ease;
}

.NavLink:hover {
  color: var(--main-accent-color);
}

.activeLink {
  color: var(--main-accent-color);
}

.activeLink::after {
  content: '';
  display: block;
  width: 100%;
  height: 2px;
  background-color: var(--main-accent-color);
}
/*# sourceMappingURL=Nav.module.css.map */
