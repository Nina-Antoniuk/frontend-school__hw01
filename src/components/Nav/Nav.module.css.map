{
    "version": 3,
    "mappings": "AAGE,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EADrC,AAAA,IAAI,CAAC;IAED,QAAQ,EAAE,KAAK;IACf,SAAS,EAAE,GAAG;IACd,MAAM,EAAE,KAAK;IACb,OAAO,EAAE,OAAO;GAEnB;;;AAED,AAAA,QAAQ,CAAC;EACP,OAAO,EAAE,CAAC;EACV,eAAe,EAAE,IAAI;CAMtB;;AAJC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,QAAQ;EAJxC,AAAA,QAAQ,CAAC;IAKL,OAAO,EAAE,IAAI;IACb,eAAe,EAAE,YAAY;GAEhC;;;AAED,AAAA,QAAQ,CAAC;EACP,OAAO,EAAE,KAAK;EACd,OAAO,EAAE,MAAM;EACf,SAAS,EAAE,IAAI;EACf,eAAe,EAAE,IAAI;EACrB,KAAK,EAAE,iBAAiB;EACxB,UAAU,EAAE,KAAK,CAAC,YAAY,CAAC,IAAI;CAKpC;;AAXD,AAQE,QARM,AAQL,MAAM,CAAC;EACN,KAAK,EAAE,wBAAwB;CAChC;;AAGH,AAAA,WAAW,CAAC;EACV,KAAK,EAAE,wBAAwB;CAShC;;AAVD,AAGE,WAHS,AAGR,OAAO,CAAC;EACP,OAAO,EAAE,EAAE;EACX,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,GAAG;EACX,gBAAgB,EAAE,wBAAwB;CAC3C",
    "sources": [
        "Nav.module.scss",
        "../../shared/scss/variables.scss"
    ],
    "names": [],
    "file": "Nav.module.css"
}