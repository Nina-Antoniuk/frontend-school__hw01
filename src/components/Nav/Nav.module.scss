@import '../../constants/scss/variables.scss';

.Nav {
  @media screen and (min-width: $breakpoint-min-tablet) {
    position: fixed;
    min-width: 20%;
    height: 100vh;
    padding: 3vh 2vw;
  }
}

.NavList {
  padding: 0;
  list-style-type: none;

  @media screen and (max-width: $breakpoint-max-mobile) {
    display: flex;
    justify-content: space-around;
  }
}

.NavLink {
  display: block;
  padding: 10px 0;
  font-size: 20px;
  text-decoration: none;
  color: var(--dark-color);
  transition: color var(--delay) ease;

  &:hover {
    color: var(--main-accent-color);
  }
}

.activeLink {
  color: var(--main-accent-color);

  &::after {
    content: '';
    display: block;
    width: 100%;
    height: 2px;
    background-color: var(--main-accent-color);
  }
}
