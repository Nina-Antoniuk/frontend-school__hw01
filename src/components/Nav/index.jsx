import React from 'react';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'consts';
import styles from './Nav.module.scss';

const Nav = function Nav() {
  return (
    <nav className={styles.Nav}>
      <ul className={styles.NavList}>
        <li>
          <NavLink
            exact
            to={ROUTES.HOME}
            className={styles.NavLink}
            activeClassName={styles.activeLink}
          >
            News
          </NavLink>
        </li>
        <li>
          <NavLink
            to={`${ROUTES.USER_PROFILE}/${'dave.xp'}`}
            className={styles.NavLink}
            activeClassName={styles.activeLink}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
export default Nav;
