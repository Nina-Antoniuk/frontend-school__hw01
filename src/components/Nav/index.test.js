import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom';
import Nav from './index';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test('render Nav', () => {
  const history = createMemoryHistory();
  render(
    <Router history={history}>
      <Nav />
    </Router>,
    container,
  );
  expect(container).toMatchSnapshot();
});

test('check for a home link', () => {
  const history = createMemoryHistory();
  render(
    <Router history={history}>
      <Nav />
    </Router>,
  );

  expect(screen.getByText(/News/i)).toBeInTheDocument();
});

test('check for a profile link', () => {
  const history = createMemoryHistory();
  render(
    <Router history={history}>
      <Nav />
    </Router>,
  );

  expect(screen.getByText(/profile/i)).toBeInTheDocument();
});
