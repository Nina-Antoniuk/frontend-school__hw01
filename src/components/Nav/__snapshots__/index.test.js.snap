// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Nav 1`] = `<div />`;
